Paintroid/src/main/java/org/catrobat/paintroid/tools/helper/JavaFillAlgorithm.kt

```diff
@@ -0,0 +1,204 @@
+/*
+ * Paintroid: An image manipulation application for Android.
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.paintroid.tools.helper
+
+import android.graphics.Bitmap
+import android.graphics.Point
+import androidx.annotation.VisibleForTesting
+import java.util.LinkedList
+import java.util.Queue
+
+private const val UP = true
+private const val DOWN = false
+
+class JavaFillAlgorithm : FillAlgorithm {
```

Comment:
Please simplify and clean the code.

Reply:
I tried cleaning it up a bit, though there wasn't too much to clean 😅

Reply:
Ok, I'll refactor some code for you ;)